cql3/statements/select_statement.cc

```diff
@@ -1144,6 +1147,38 @@ query::partition_slice indexed_table_select_statement::get_partition_slice_for_g
             }
 
             partition_slice_builder.with_ranges(clustering_restrictions->bounds_ranges(options));
+        } else if (has_token(_restrictions->get_partition_key_restrictions()->expression) && is_token_v2_index) {
```

Comment:
Just to continue that thought (again no action needed), at prepare time we could choose which path to take by selecting a sub-class of indexed_table_select_statement, rather than this run-time choice.